From 6219dd86dbddad06e41915e1f4262dec6f74d2cf Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Fri, 27 Apr 2018 00:04:21 +0100
Subject: [PATCH 06/15] msm8953-common: List additional native libraries to
 expose to apps

 * Initially based on the list defined by Xiaomi, trimmed to include
   only the vendor libs that are also being shipped by us.

 * Example bellow of a failure before one lib was made accessible to
   a camera app.

   Before exposing:
     I CAM_HexagonEnv: Will attempt to load libadsprpc.so from the system partition
     I CAM_HexagonEnv: Loading libhalide_hexagon_host.so from the app
     I CAM_CamBoxHelper: Computing layout for window: 1080x1920, and preview: 3000x4000, orientation: PORTRAIT, isMultiWindow: false
     I CAM_CamBoxHelper: placing extra widgets above the bottom bar
     I CAM_CamBoxHelper: Computed layout: CameraLayoutBoxes{window=1080x1920, preview=Rect(0, 106 - 1080, 1546), uncoveredPreview=Rect(0, 173 - 1080, 1546), topBar=Rect(0, 0 - 1080, 106), optionsBar=Rect(0, 0 - 1080, 173), bottomBar=Rect(0, 1546 - 1080, 1920), extraBottomWidgets=Rect(0, 1488 - 1080, 1546)}
     W CAM_HexagonEnv: Failed to load Hexagon library: java.lang.UnsatisfiedLinkError: dlopen failed: library "libadsprpc.so" not found

   After exposing:
     I CAM_HexagonEnv: Will attempt to load libadsprpc.so from the system partition
     I CAM_HexagonEnv: Loading libhalide_hexagon_host.so from the app
     I CAM_CamBoxHelper: Computed layout: CameraLayoutBoxes{window=1078x1860, preview=Rect(0, 0 - 1078, 808), uncoveredPreview=Rect(0, 173 - 1078, 1437), topBar=Rect(0, 0 - 1078, 106), optionsBar=Rect(0, 0 - 1078, 173), bottomBar=Rect(0, 1437 - 1078, 1860), extraBottomWidgets=Rect(0, 1437 - 1078, 1495)}
     I CAM_HexagonEnv: Writing libhalide_hexagon_remote_skel.so to /data/user/0/<package_name_removed>/libhalide_hexagon_remote_skel.so
     I CAM_HexagonEnv: ADSP_LIBRARY_PATH=/data/user/0/<package_name_removed>;/dsp

Change-Id: I552dc44a32221d2d3cec48e1c64ed72c470d412e
---
 configs/public.libraries.txt | 3 +++
 msm8953.mk                   | 4 ++++
 2 files changed, 7 insertions(+)
 create mode 100644 configs/public.libraries.txt

diff --git a/configs/public.libraries.txt b/configs/public.libraries.txt
new file mode 100644
index 0000000..1b8eb7e
--- /dev/null
+++ b/configs/public.libraries.txt
@@ -0,0 +1,3 @@
+libadsprpc.so
+libOpenCL.so
+libqti-perfd-client.so
diff --git a/msm8953.mk b/msm8953.mk
index fdfa187..ca15897 100644
--- a/msm8953.mk
+++ b/msm8953.mk
@@ -73,6 +73,10 @@ PRODUCT_COPY_FILES += \
     frameworks/native/data/etc/android.software.sip.voip.xml:$(TARGET_COPY_OUT_VENDOR)/etc/permissions/android.software.sip.voip.xml \
     frameworks/native/data/etc/handheld_core_hardware.xml:$(TARGET_COPY_OUT_VENDOR)/etc/permissions/handheld_core_hardware.xml
 
+# Additional native libraries
+PRODUCT_COPY_FILES += \
+    $(LOCAL_PATH)/configs/public.libraries.txt:$(TARGET_COPY_OUT_VENDOR)/etc/public.libraries.txt
+
 # ANT
 PRODUCT_PACKAGES += \
     AntHalService \
-- 
2.7.4

